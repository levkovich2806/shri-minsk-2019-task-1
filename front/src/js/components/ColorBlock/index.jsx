import React, { PureComponent } from "react";
import styles from "./index.module.scss";

class ColorBlock extends PureComponent {
  render() {
    const {
      color: { color = "orange", id },
      changeStatus,
      checked = false
    } = this.props;
    return (
      <label className={styles.container}>
        <input
          type="checkbox"
          defaultChecked={checked}
          onClick={() => changeStatus(id)}
        />
        <span className={styles.checkmark} style={{ backgroundColor: color }} />
      </label>
    );
  }
}

export default ColorBlock;
