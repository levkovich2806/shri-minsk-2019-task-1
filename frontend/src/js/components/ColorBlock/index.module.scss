.container {
  position: relative;
  height: 32px;
  width: 32px;
  margin-left: 16px;
  cursor: pointer;
  line-height: 22px;
  font-size: 18px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.container input {
  position: absolute;
  opacity: 0;
  cursor: pointer;
  height: 0;
  width: 0;
}

.checkmark {
  opacity: 0.4;
  position: absolute;
  height: 32px;
  width: 32px;
  background-color: #d9d9d9;
  border-radius: 8px;
  transition: opacity 0.5s;
}

.container:hover input ~ .checkmark {
  opacity: 1;
}

.container:active input ~ .checkmark {
  opacity: 0.8;
}

.checkmark::after {
  content: "";
  position: absolute;
  display: none;
}

.container .checkmark::after {
  left: 10px;
  top: 3px;
  width: 8px;
  height: 18px;
  border: solid black;
  border-width: 0 3px 3px 0;
  -webkit-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
}

.container input:checked ~ .checkmark::after {
  display: block;
}

.container input:checked ~ span {
  opacity: 1;
}

@media (max-width: 500px) {
  .container {
    width: 48px;
    height: 48px;
    flex: 0 0 auto;
    margin-right: 3px;
  }

  .checkmark {
    width: 48px;
    height: 48px;
  }

  .container .checkmark::after {
    left: 16px;
    top: 3px;
    width: 14px;
    height: 28px;
  }
}
