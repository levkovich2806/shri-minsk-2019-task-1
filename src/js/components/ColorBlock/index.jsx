import React, { PureComponent } from "react";
import styles from "./index.module.scss";

class ColorBlock extends PureComponent {
  render() {
    const { color = "orange" } = this.props;
    return <div className={styles.block} style={{ backgroundColor: color }} />;
  }
}

export default ColorBlock;
