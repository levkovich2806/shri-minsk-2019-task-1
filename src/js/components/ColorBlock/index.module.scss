.block {
  width: 32px;
  height: 32px;
  opacity: 0.4;
  margin-left: 10px;
  border-radius: 8px;
  cursor: pointer;
  transition: opacity 0.5s;
}

.block:hover {
  opacity: 1;
}

.block:active {
  opacity: 0.8;
}

@media (max-width: 500px) {
  .block {
    width: 48px;
    height: 48px;
    flex: 0 0 auto;
    margin-right: 3px;
  }
}
