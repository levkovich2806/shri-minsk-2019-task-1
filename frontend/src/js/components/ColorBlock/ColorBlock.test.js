import React from "react";
import { mount } from "enzyme";
import ColorBlock from "./index.jsx";
import renderer from "react-test-renderer";

describe("ColorBlock", function () {
  test("ColorBlock rendering", () => {
    const params = {
      color: {
        color: "#000",
        id: 0
      }
    };
    const component = renderer.create(<ColorBlock {...params} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("ColorBlock check background with color", () => {
    const params = {
      color: {
        color: "#000",
        id: 0
      }
    };
    const wrapper = mount(<ColorBlock {...params} />);

    const span = wrapper.find("span.checkmark");
    let containerStyle = span.get(0).props.style["backgroundColor"];
    expect(containerStyle).toBe("#000");
  });

  test("ColorBlock check background without send color", () => {
    const params = {
      color: {
        id: 0
      }
    };
    const wrapper = mount(<ColorBlock {...params} />);

    const span = wrapper.find("span.checkmark");
    let containerStyle = span.get(0).props.style["backgroundColor"];
    expect(containerStyle).toBe("orange");
  });

  it('ColorBlock handle click', () => {
    const handleClick = jest.fn();
    const wrapper = mount(<ColorBlock changeStatus={handleClick} color={{ id: 0 }} />);
    const checkbox = wrapper.find('input[type="checkbox"]');
    checkbox.simulate('click');
    expect(handleClick).toHaveBeenCalled();
  });
});
