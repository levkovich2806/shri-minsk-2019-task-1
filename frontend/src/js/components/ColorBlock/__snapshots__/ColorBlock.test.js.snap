// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorBlock ColorBlock rendering 1`] = `
<label
  className="container"
>
  <input
    defaultChecked={false}
    onClick={[Function]}
    type="checkbox"
  />
  <span
    className="checkmark"
    style={
      Object {
        "backgroundColor": "#000",
      }
    }
  />
</label>
`;
